### PR TITLE
Refactor StadiaTarget to SshTarget

### PR DESCRIPTION
**Pull request by antonrohr**
and remove then unused code. This includes the classes
* ConnectToStadiaWidget
* RetrieveInstancesWidget
* RetrieveInstances

It also removes ConnectToStadiaWidget related code in SessionSetupDialog

Please note, there are locations where an implementation is missing, this will be added in subsequent PRs.

### CODE REVIEW

File: src/SessionSetup/include/SessionSetup/Connections.h [hunk 1]
```diff
@@ -52,26 +52,25 @@ class SshConnectionArtifacts {
 };
 
 /*
- * The class StadiaConnection describes an active connection to a stadia instance. This class holds
- * an Instance object it is connected. The ServiceDeployManager which carries the active connection
- * and the ssh tunnel. And the grpc channel that is used for the communication with the instance.
- * This class is meant to be constructed and then not modified anymore. Only ConnectToStadiaWidget
- * is allowed to modify the members, which is used to move out members for reusing them.
+ * The class SshConnection describes an active connection to a machine via ssh. This class holds
```
Comment (florian-kuebler) [inline src/SessionSetup/include/SessionSetup/Connections.h:55]:
```suggestion
 * The class `SshConnection` describes an active connection to a machine via ssh. This class holds
```

File: src/SessionSetup/include/SessionSetup/Connections.h [hunk 2]
```diff
@@ -52,26 +52,25 @@ class SshConnectionArtifacts {
...
+ * a AddrAndPort which is the target of the ssh connection. The ServiceDeployManager which carries
```
Comment (florian-kuebler) [inline src/SessionSetup/include/SessionSetup/Connections.h:56]:
```suggestion
 * a `AddrAndPort` which is the target of the ssh connection. The `ServiceDeployManager` which carries
```

File: src/SessionSetup/include/SessionSetup/Connections.h [hunk 3]
```diff
@@ -52,26 +52,25 @@ class SshConnectionArtifacts {
...
+ * the active connection and the ssh tunnel. And the grpc channel that is used for the communication
+ * with the machine. This class is meant to be constructed and then not modified anymore. Only
+ * SshConnectionWidget is allowed to modify the members, which is used to move out members for
```
Comment (florian-kuebler) [inline src/SessionSetup/include/SessionSetup/Connections.h:59]:
```suggestion
 * `SshConnectionWidget` is allowed to modify the members, which is used to move out members for
```

File: src/SessionSetup/include/SessionSetup/Connections.h (hunk 1 above)
Comment (florian-kuebler) [reply src/SessionSetup/include/SessionSetup/Connections.h:55]: This helps IDEs to rename classnames in comments when renaming the class.

File: src/SessionSetup/include/SessionSetup/TargetConfiguration.h [hunk 4]
```diff
@@ -14,32 +14,32 @@
 namespace orbit_session_setup {
 
 /*
- * StadiaTarget describes a successful connection to an instance and a selected process. The
+ * SshTarget describes a successful ssh connection to a machine and a selected process. The
```
Comment (florian-kuebler) [inline src/SessionSetup/include/SessionSetup/TargetConfiguration.h:17]:
```suggestion
 * `SshTarget` describes a successful ssh connection to a machine and a selected process. The
```

File: src/SessionSetup/include/SessionSetup/TargetConfiguration.h [hunk 5]
```diff
@@ -14,32 +14,32 @@
...
  * class is built in SessionSetupDialog and mainly used in OrbitMainWindow. This class is meant
```
Comment (florian-kuebler) [inline src/SessionSetup/include/SessionSetup/TargetConfiguration.h:18]:
```suggestion
 * class is built in `SessionSetupDialog` and mainly used in `OrbitMainWindow`. This class is meant
```

File: src/SessionSetup/include/SessionSetup/TargetConfiguration.h [hunk 6]
```diff
@@ -14,32 +14,32 @@
...
  * to be constructed and then not modified anymore. Only SessionSetupDialog is allowed to modify
```
Comment (florian-kuebler) [inline src/SessionSetup/include/SessionSetup/TargetConfiguration.h:19]:
```suggestion
 * to be constructed and then not modified anymore. Only `SessionSetupDialog` is allowed to modify
```

**Review by florian-kuebler — APPROVED** [review]
Only super nit comments about comments. LGTM. Thanks!

**Review by beckerhe — APPROVED** [review]
Nice. Thank you!

**Pull request merged**